docs/guide/extending_online_editor.md

```diff
@@ -255,6 +255,99 @@ In the example above, the template files for the front end could be:
 
 Templates for Content View in the Back Office would be `app/Resources/views/themes/admin/field_type/ezrichtext/custom_style/highlighted_word.html.twig` and `app/Resources/views/themes/admin/field_type/ezrichtext/custom_style/highlighted_block.html.twig` respectively (assuming Admin SiteAccess uses the `admin` theme).
 
+## Custom data attributes and classes
+
+You can add custom data attributes and CSS classes to elements in the Online Editor.
+
+The available elements are:
+
+- `embedinline`
+- `embed`
+- `formatted`
+- `heading`
+- `embedimage`
+- `ul`
+- `ol`
+- `li`
+- `paragraph`
+- `table`
+- `tr`
+- `td`
+
+### Custom data attributes
+
+Custom data attributes are configured under the `fieldtypes.ezrichtext.attributes` key.
+The configuration is SiteAccess-aware.
+
+A custom data attribute can belong to one of four types: `choice`, `boolean`, `string`, or `number`.
+You can also set each attribute to be `required` and set its `default_value`.
+
+For the `choice` type, you must provide an array of available `choices`.
+Adding `multiple` enables you to choose whether more than one option can be selected.
+
+The example below adds two data attributes, `custom_attribute` and `another_attribute`
+to the Heading element:
+
+``` yaml
+ezpublish:
+    system:
+        # The configuration only works with an admin (Back Office) SiteAccess
+        <siteaccess>:
+            fieldtypes:
+                ezrichtext:
+                    attributes:
+                        heading:
+                            custom-attribute:
+                                type: boolean
+                                default_value: false
+                            another-attribute:
+                                type: choice
+                                choices: [attr1, attr1]
+                                default_value: attr2
+                                required: false
+                                multiple: true
+```
+
+This configuration will output `data-<attribute_name>` in the corresponding HTML element,
+in this example as `data-custom-attribute` and `data-another-attribute`.
```

Comment:
Actually, I've completely missed this.
For a paragraph if e.g. `value1` was chosen for `my-attribute` custom data attribute, a HTML output would be:
```html
<p data-ezattribute-my-attribute="value1"></p>
```

It might be worth to mention that in case of data attributes, if someone allowed multiple options, e.g. `value1` and `value2` to be chosen, the the rendered HTML for e.g. `my-attribute` would be e.g.:
```html
<p data-ezattribute-my-attribute="value1,value2"></p>
```
It's comma-separated on purpose and need to be handled (e.g split into values on front-end) by a developer if he allowed multiple choices.